ci/tests/vmachine/vmachine_test.py

```diff
@@ -135,103 +136,72 @@ def check_scrubbing_test():
-                                                  'model=e1000 --import'.format(machine_name, vpool_name, disk_name))
-            if err:
-                GeneralVMachine.logger.error("Error while creating vmachine: {0}".format(err))
-
-        def snapshot_vdisks():
-            vds = GeneralVDisk.get_vdisks()
-            for disk in vds:
-                metadata = {'label': 'snap-' + disk.name,
-                            'is_consistent': True,
-                            'timestamp': time.time(),
-                            'is_automatic': False,
-                            'is_sticky': False}
-                snapshot_id = VDiskController.create_snapshot(disk.guid, metadata)
-                GeneralVMachine.logger.info("Created snapshot {0} for vdDisk {1}".format(snapshot_id, disk.name))
-
-        # snapshotting disks for the first time
-        snapshot_vdisks()
-        counter = timeout / timer_step
+        disk_name = "scrubdisk"
+        GeneralVMachine.logger.info("Starting RAW disk creation")
+        out, err, _ = General.execute_command('qemu-img convert -O raw {0}{1} /mnt/{2}/{3}.raw'.format(template_folder, image_name, vpool_name, disk_name))
+        if err:
+            GeneralVMachine.logger.error("Error while creating raw disk: {0}".format(err))
+
+        def snapshot_vdisk(vdisk):
+            metadata = {'label': 'snap-' + vdisk.name,
+                        'is_consistent': True,
+                        'timestamp': time.time(),
+                        'is_automatic': False,
+                        'is_sticky': False}
+            VDiskController.create_snapshot(vdisk.guid, metadata)
+
+        counter = initial_counter
+        while counter and vdisk is None:
+            time.sleep(step)
+            vdisk = VDiskList.get_by_devicename_and_vpool('/' + disk_name + '.raw', vpool)
+            counter -= step
+        assert counter > 0, "Vdisk with name {0} didn't appear in the model after 60 seconds".format(disk_name)
+        # snapshot disks for the first time
+        snapshot_vdisk(vdisk)
+        counter = initial_counter
         while counter > 0:
-            time.sleep(timer_step)
-            counter -= 1
-            snapshot_vdisks()
+            time.sleep(step)
+            out, err, _ = General.execute_command('dd if=/dev/zero of=/mnt/{0}/{1}.raw bs=10K count=1000 conv=notrunc'.format(vpool_name, disk_name))
+            counter -= step
+            snapshot_vdisk(vdisk)
 
-        # stopping machines
-        vms = hypervisor.sdk.get_vms()
-        for vm in vms:
-            GeneralVMachine.logger.info("Stopping {0} vmachine".format(vm.name()))
-            out, err, _ = General.execute_command('virsh destroy {0}'.format(vm.name()))
-            if err:
-                GeneralVMachine.logger.error("Error while stopping vmachine: {0}".format(err))
+        vdisk = VDiskList.get_by_devicename_and_vpool('/' + disk_name + '.raw', vpool)
```

Comment:
Not required perse, you have the ```vdisk``` from the above fetches, and if I'm correct, it's the same vdisk.